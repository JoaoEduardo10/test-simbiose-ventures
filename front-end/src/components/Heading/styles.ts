import { css, styled } from "styled-components";

export const HeadingConteiner = styled.h1`
  ${() => css`
    text-align: center;
    padding-top: 1rem;
    margin-bottom: 3rem;
    text-transform: uppercase;
  `}
`;
