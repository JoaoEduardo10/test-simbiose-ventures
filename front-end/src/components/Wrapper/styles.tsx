import { css, styled } from "styled-components";

export const WrapperConteiner = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.light_gray};
    width: 100%;
    height: 100vh;
  `}
`;
