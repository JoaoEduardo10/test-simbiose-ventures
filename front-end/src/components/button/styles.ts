import { css, styled } from "styled-components";

export const ButtonConteiner = styled.button`
  ${({ theme }) => css`
    border: none;
    background-color: ${theme.colors.blue};
    padding: 1rem;
    color: ${theme.colors.secondary_color};
    font-size: 1.4rem;
    border-radius: 0.5rem;
    cursor: pointer;
    transition: all 300ms ease-in-out;

    &:hover {
      background-color: ${theme.colors.secondary_color};
      color: ${theme.colors.blue};
      border: 0.1rem solid ${theme.colors.blue};
    }
  `}
`;
