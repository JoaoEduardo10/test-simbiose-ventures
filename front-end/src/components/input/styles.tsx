import { css, styled } from "styled-components";

export const InputConteiner = styled.div`
  ${() => css`
    display: flex;
    flex-direction: column;
    gap: 0.1rem;
    width: 90%;
  `}
`;

export const Label = styled.label<{ $touchedLabel: boolean }>`
  ${({ $touchedLabel }) => css`
    left: 0;
    top: 0;
    font-size: 1.4rem;
    ${$touchedLabel ? "display: none;" : "display: block;"}
    text-transform: capitalize;
    cursor: pointer;
  `}
`;

export const Input = styled.input<{ $touchedInput: boolean }>`
  ${({ theme, $touchedInput }) => css`
    height: 3rem;
    width: 23rem;
    outline: none;
    padding: 0 1rem;
    border-radius: 0.5rem;
    border: none;
    border: 0.1rem solid
      ${!$touchedInput ? theme.colors.primary_color : "transparent"};
  `}
`;
