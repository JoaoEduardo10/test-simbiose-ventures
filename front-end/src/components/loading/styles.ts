import { css, styled } from "styled-components";

export const LoadingConteiner = styled.button`
  ${() => css`
    position: fixed;
    width: 100%;
    height: 100%;
    z-index: 2;
    background-color: rgba(0, 0, 0, 0.6);
    display: flex;
    justify-content: center;
    align-items: center;
    left: 0;
    top: 0;
  `}
`;

export const CicleOne = styled.div`
  ${({ theme }) => css`
    width: 5rem;
    height: 5rem;
    border-radius: 50%;
    border: 0.4rem solid ${theme.colors.blue};
    border-top: 0.4rem solid transparent;
    display: flex;
    justify-content: center;
    align-items: center;
    animation: cicleOne infinite ease-in-out 1s;

    @keyframes cicleOne {
      0% {
        transform: rotate(360deg);
      }

      100% {
        transform: translate(0deg);
      }
    }
  `}
`;

export const CicleTwo = styled.div`
  ${() => css`
    width: 3rem;
    height: 3rem;
    border-radius: 50%;
    border: 0.2rem solid green;
    border-bottom: 0.2rem solid transparent;
    animation: cicleTwo infinite ease-in-out reverse 300ms;

    @keyframes cicleTwo {
      0% {
        transform: rotate(360deg);
      }

      100% {
        transform: translate(0deg);
      }
    }
  `}
`;
