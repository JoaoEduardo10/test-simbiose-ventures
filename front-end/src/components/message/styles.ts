/* eslint-disable indent */
import { css, styled } from "styled-components";

export type MessageConteinerProps = {
  $type: "error" | "ok";
  $state: boolean;
};

const show = (state: MessageConteinerProps["$state"]) => css`
  ${state == true
    ? css`
        transform: translateX(50%);
        opacity: 1;
        animation: keyShow ease-in-out 300ms;

        @keyframes keyShow {
          0% {
            transform: translateX(-100%);
            opacity: 0;
          }
          100% {
            transform: translateX(0);
            opacity: 1;
          }
        }
      `
    : css`
        transform: translateX(-100%);
        opacity: 0;
        animation: keyShowFalse ease-in-out 1s;

        @keyframes keyShowFalse {
          0% {
            transform: translateX(0%);
            opacity: 1;
          }

          50% {
            transform: translateX(100%);
          }

          100% {
            transform: translateX(-100%);
            opacity: 0;
          }
        }
      `}
`;

export const MessageConteiner = styled.div<MessageConteinerProps>`
  ${({ theme, $type, $state }) => css`
    position: fixed;
    bottom: 20%;
    left: 2%;
    width: 35rem;
    height: 8rem;
    border: none;
    padding: 1rem;
    font-size: 1.4rem;
    background-color: ${theme.colors.secondary_color};
    color: ${theme.colors.blue};
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.6rem;
    border: ${$type == "ok" ? "0.1rem solid green" : "0.1rem solid red"};

    ${show($state)}
  `}
`;
