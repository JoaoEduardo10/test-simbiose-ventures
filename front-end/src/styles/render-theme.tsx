import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { GlobalStyles } from "./global-styles";
import { theme } from "./theme";

export const RenderTheme = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
};
