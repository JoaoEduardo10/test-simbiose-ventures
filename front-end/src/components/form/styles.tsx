import { css, styled } from "styled-components";

import { ButtonConteiner } from "../button/styles";

export const FormConteiner = styled.form`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondary_color};
    display: flex;
    gap: 1rem;
    margin: 0 auto;
    width: 60%;
    height: 9rem;
    border-radius: 1rem;
    align-items: center;
    padding: 0 1.6rem;
    border: 0.1rem solid ${theme.colors.primary_color};
    box-shadow: 0px 2px 55px -31px rgba(0, 0, 0, 0.75);

    ${ButtonConteiner} {
      margin-top: 1.2rem;
    }
  `}
`;
