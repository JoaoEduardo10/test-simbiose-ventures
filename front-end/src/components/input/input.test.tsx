/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, expect, it, vi } from "vitest";
import { RenderTheme } from "../../styles/render-theme";
import { InputComponent } from ".";
import { fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import "jest-styled-components";
import { theme } from "../../styles/theme";

describe("<InputComponent />", () => {
  const handleValue = vi.fn();

  it("should render input to form", () => {
    RenderTheme(
      <InputComponent
        handleValue={handleValue}
        value="name"
        text="name"
        type="text"
      />
    );

    const input = screen.getByPlaceholderText("name");

    expect(input).toBeInTheDocument();
    expect(input).toHaveStyleRule(
      "border",
      `0.1rem solid ${theme.colors.primary_color}`
    );
  });

  it("should render the input to view the value", () => {
    RenderTheme(
      <InputComponent
        handleValue={handleValue}
        value="name"
        text="name"
        type="text"
        readOnly={true}
        touchedLabe={true}
      />
    );

    const input = screen.getByPlaceholderText("name") as any;

    expect(input.value).toBe("name");
    expect(input).toHaveStyleRule("border", "0.1rem solid transparent");
  });

  it("should write to input", () => {
    RenderTheme(
      <InputComponent
        handleValue={handleValue}
        value="name"
        text="name"
        type="text"
      />
    );

    const input = screen.getByPlaceholderText("name") as any;

    expect(handleValue).toHaveBeenCalledWith("");

    fireEvent.change(input, { target: { value: "test" } });

    expect(handleValue).toHaveBeenCalledWith("test");
  });

  it("should to match snapshot ", () => {
    const { container } = RenderTheme(
      <InputComponent
        handleValue={handleValue}
        value="name"
        text="name"
        type="text"
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
