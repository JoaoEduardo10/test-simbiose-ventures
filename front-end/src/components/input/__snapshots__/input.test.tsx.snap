// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<InputComponent /> > should to match snapshot  1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 0.1rem;
  width: 90%;
}

.c1 {
  left: 0;
  top: 0;
  font-size: 1.4rem;
  display: block;
  text-transform: capitalize;
  cursor: pointer;
}

.c2 {
  height: 3rem;
  width: 23rem;
  outline: none;
  padding: 0 1rem;
  border-radius: 0.5rem;
  border: none;
  border: 0.1rem solid #222;
}

<div
  aria-label="input"
  class="c0"
>
  <label
    class="c1"
    for="text"
  >
    name
  </label>
  <input
    class="c2"
    id="text"
    placeholder="name"
    type="text"
    value="name"
  />
</div>
`;
